compiler/src/Language/Bond/Codegen/Cpp/ApplyOverloads.hs

```diff
@@ -22,20 +23,22 @@ data Protocol =
 
 
 -- Apply overloads
-applyOverloads :: [Protocol] -> Text -> Text -> Declaration -> Text
-applyOverloads protocols attr body Struct {..} | null declParams = [lt|
+applyOverloads :: [Protocol] -> MappingContext -> Text -> Text -> Declaration -> Text
+applyOverloads protocols cpp attr body s@Struct {..} | null declParams = [lt|
     //
     // Overloads of Apply function with common transforms for #{declName}.
     // These overloads will be selected using argument dependent lookup
     // before bond::Apply function templates.
     //
-    #{attr}bool Apply(const bond::To<#{declName}>& transform,
-               const bond::bonded<#{declName}>& value)#{body}
+    #{attr}bool Apply(const bond::To< #{qualifiedName}>& transform,
```

Comment:
Minor: we don't need to add spaces here to deal with things like `<<` and `>>`. We have a fix-up step that adds spaces where needed (and handles things like trigraphs).